src/Utils/TempDirectoryCreator.php

```diff
@@ -9,29 +9,40 @@
 
 namespace Infection\Utils;
 
+use Infection\Filesystem\Filesystem;
+
 class TempDirectoryCreator
 {
+    const BASE_DIR_NAME = 'infection';
+
+    /**
+     * @var Filesystem
+     */
+    private $fs;
+
     /**
      * @var string
      */
-    private $tempDirectory;
+    private $tempDirectoryPath;
```

Comment:
as per the comment above, `tmpDir`? eventually `*Path` but as there is no `Directory` there is no confusing IMO. I would as a result change the name of the class as well